src/middlewared/middlewared/plugins/account.py

```diff
@@ -27,50 +81,477 @@ class UserService(CRUDService):
                 pass
         return user
 
-    async def do_update(self, id, data):
-
-        user = await self.middleware.call('datastore.query', 'account.bsdusers', [('id', '=', id)], {'prefix': 'bsdusr_'})
-        if not user:
-            raise CallError(f'User {id} does not exist', errno.ENOENT)
-        user = user[0]
-
-        if 'sshpubkey' in data:
-            keysfile = f'{user["home"]}/.ssh/authorized_keys'
-            pubkey = data.pop('sshpubkey')
-            if pubkey is None:
-                if os.path.exists(keysfile):
-                    try:
-                        os.unlink(keysfile)
-                    except OSError:
-                        pass
+    @accepts(Dict(
+        'user_create',
+        Int('uid'),
+        Str('username', required=True),
+        Int('group'),
+        Bool('group_create', default=False),
+        Str('home', default='/nonexistent'),
+        Str('home_mode', default='755'),
+        Str('shell', default='/bin/csh'),
+        Str('full_name', required=True),
+        Str('email'),
+        Str('password'),
+        Bool('password_disabled', default=False),
+        Bool('locked', default=False),
+        Bool('microsoft_account', default=False),
+        Bool('sudo', default=False),
+        Str('sshpubkey'),
+        List('groups'),
+        Dict('attributes', additional_attrs=True),
+        register=True,
+    ))
+    async def do_create(self, data):
+
+        verrors = ValidationErrors()
+
+        if (
+            not data.get('group') and not data.get('group_create')
+        ) or (
+            data.get('group') is not None and data.get('group_create')
+        ):
+            verrors.add('group', f'You need to either provide a group or group_create', errno.EINVAL)
+
+        await self.__common_validation(verrors, data)
+
+        if data.get('sshpubkey') and not data['home'].startswith('/mnt'):
+            verrors.add('sshpubkey', 'Home directory is not writable, leave this blank"')
```

Comment:
This error message may confuse user. Maybe this check should be moved down (after homedir creation), with different condition?